daemon/daemon.go

```diff
@@ -683,6 +683,14 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 	if err != nil {
 		return nil, fmt.Errorf("error initializing graphdriver: %v", err)
 	}
+	// make sure the graphdriver actually works as configured
+	// often if there is an unknown backingfs or dangling mounts from an improperly shutdown daemon there can be issues
+	logrus.Debugf("testing graph driver initialization")
+	if err := testGraphDriver(driver); err != nil {
+		driver.Cleanup()
+		return nil, fmt.Errorf("error initializing graphdriver: %v\nDriver Status:\n", err, driver.Status())
```

Comment:
Cool, I'll update, and have to update the test too, b/c it's not finding errors when I expect it to.